.github/workflows/mirror-tooling-images.yml

```diff
@@ -0,0 +1,59 @@
+name: "[Daily] Mirror Tooling Images to GHCR"
+
+on:
+  schedule:
+    - cron: "0 3 * * *" # every day at 3 AM UTC
+  workflow_dispatch:
+    inputs:
+      ghcr_namespace:
+        description: "GHCR Namespace"
+        required: true
+        default: "ghcr.io/project-copacetic/copacetic"
+
+env:
+  GHCR_NAMESPACE: ${{ github.event.inputs.ghcr_namespace || 'ghcr.io/project-copacetic/copacetic' }}
+
+jobs:
+  mirror:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+
+    strategy:
+      matrix:
+        images:
+          - "debian:10-slim"
+          - "debian:11-slim"
+          - "debian:12-slim"
+          - "debian:stable-slim"
+          - "ubuntu:20.04"
+          - "ubuntu:22.04"
+          - "ubuntu:latest"
+          - "mcr.microsoft.com/azurelinux/base/core:3.0"
+          - "mcr.microsoft.com/cbl-mariner/base/core:2.0"
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Log in to GHCR
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
+
+      - name: Pull and push cache image
+        env:
+          SOURCE_IMAGE: ${{ matrix.images }}
+        run: |
+          if [[ "$SOURCE_IMAGE" == */* ]]; then
+            # Image with registry hostname (strip hostname)
+            TARGET_IMAGE=$(echo "$SOURCE_IMAGE" | sed 's#^[^/]*/##')
+          else
+            # Official library images (no registry hostname)
+            TARGET_IMAGE="$SOURCE_IMAGE"
+          fi
+
+          echo "Processing image: $SOURCE_IMAGE -> $GHCR_NAMESPACE/$TARGET_IMAGE"
+
+          docker pull "$SOURCE_IMAGE"
```

Comment:
can we prefix with mirror.gcr.io so it uses the google pull through cache? otherwise, we'll get throttled here

Reply:
good point, we wouldn't want the github action itself to get throttled. I'll update! presumably we can keep the mcr images as is and not worry about throttling?

Reply:
yea i was debating that, let's keep it to be consistent and in case mcr introduces throttling later?

Reply:
it seems the mcr images are not on gcr or I couldn't find them at least, do you know if they are?